tiledb/sm/misc/integral_type_casts.h

```diff
@@ -99,11 +99,6 @@ void safe_integral_cast_to_datatype(
     throw std::invalid_argument("Datatype must be integral");
   }
 
-  if (type == Datatype::BLOB) {
-    throw std::invalid_argument(
-        "Datatype::BLOB not supported in integral conversion");
-  }
-
```

Comment:
Optional: Below when the logic error is thrown it would be nice to use the `datatype_str` function to say what datatype does not support integral conversion.